### PR TITLE
feat: remove no bind rule and edit file name to include .tsx files

### PR DESCRIPTION
**Pull request by ShadyMohammed**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,57 +157,57 @@ import Footer from './Footer';
...
-4.1 Avoid extra closing tags for components without children ([self-closing-comp](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/self-closing-comp.md)) 
-   
+
+4.1 Avoid extra closing tags for components without children ([self-closing-comp](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/self-closing-comp.md))
+
 ```javascript
 // bad
-var HelloJohn = <Hello name="John"></Hello>;
-    
+var HelloJohn = <Hello name="John" />;
```
Comment (alihassan0) [inline README.md]: auto format backfired here :joy:

File: README.md [hunk 2]
```diff
@@ -152,57 +157,57 @@ import Footer from './Footer';
...
+```javascript
+// bad
 var x = function() {
-  return <Say
-    firstName="John"
-    lastName="Smith"
-         >
-    Hello
-         </Say>;
+  return (
```
Comment (alihassan0) [inline README.md]: revert theses changes as well

File: README.md [hunk 3]
```diff
@@ -299,34 +307,30 @@ var x = function() {
...
-<Hello name={ firstname } />;
-<Hello name={ firstname} />;
-<Hello name={firstname } />;
-<Hello name={
-  firstname
-} />;
-<Hello name = {firstname} />;
-<Hello name ={firstname} />;
-<Hello name= {firstname} />;
+<Hello name={firstname} />;
```
Comment (bahaagalal) [inline README.md]: Revert back to original content as this emphasizes a bad example of spaces around curly spaces and equal signs.

File: README.md [hunk 4]
```diff
@@ -299,34 +307,30 @@ var x = function() {
...
 // good
 <Hello name={firstname} />;
 <Hello name={{ firstname: 'John', lastname: 'Doe' }} />;
 <Hello>{firstname}</Hello>;
-<Hello>{ firstname }</Hello>;
-<Hello>{
-  firstname
-}</Hello>;
+<Hello>{firstname}</Hello>;
+<Hello>{firstname}</Hello>;
```
Comment (bahaagalal) [inline README.md]: revert back to original example.

File: rules/jsx.js [hunk 5]
```diff
@@ -119,25 +119,6 @@ module.exports = {
...
 
-    /**
-     * No .bind() or Arrow Functions in JSX Props.
-     * A bind call or arrow function in a JSX prop will create a brand new function on every
-     * single render. This is bad for performance, as it may cause unnecessary re-renders if
-     * a brand new function is passed as a prop to a component that uses reference equality
-     * check on the prop to determine if it should update.
-     * https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md
-     */
-    "react/jsx-no-bind": [
```
[the remark this answers is not in the document]
Comment (ShadyMohammed) [reply rules/jsx.js]: It leads to bad DX. we don't bind inline and we use arrow functions.
please check: https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578

Also if I want to access a prop in the callback I should transform the component into class component to do that.

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,8 +1,7 @@
 version: 2
 references:
-  workdir: &workdir
-    /var/code
-  
+  workdir: &workdir /var/code
+
   attach_workdir: &attach_workdir
     attach_workspace:
       at: *workdir
```
Comment (bahaagalal) [inline .circleci/config.yml]: update node to version 12.4.0

File: .dockerignore [hunk 7]
```diff
@@ -1,2 +1,167 @@
...
+
+# Windows Installer files
+*.cab
+*.msi
+*.msix
+*.msm
+*.msp
+
+# Windows shortcuts
+*.lnk
```
Comment (bahaagalal) [inline .dockerignore]: new line at the end of the file.

File: .gitignore [hunk 8]
```diff
@@ -1,38 +1,167 @@
...
+
+# Windows Installer files
+*.cab
+*.msi
+*.msix
+*.msm
+*.msp
+
+# Windows shortcuts
+*.lnk
```
Comment (bahaagalal) [inline .gitignore]: new line at the end of the file.

File: package.json [hunk 9]
```diff
@@ -29,10 +33,10 @@
...
-    "babel-eslint": "^10.0.1",
-    "eslint": "^5.16.0",
+    "babel-eslint": "^10.0.2",
+    "eslint": "^6.1.0",
     "eslint-config-crowdanalyzer": "^1.4.1",
-    "eslint-plugin-react": "^7.12.4",
+    "eslint-plugin-react": "^7.14.3",
     "npm-check": "^5.9.0"
   },
   "peerDependencies": {
```
Comment (bahaagalal) [inline package.json]: update `eslint` peer dependency to be `< 7`

File: package.json [hunk 10]
```diff
@@ -29,10 +33,10 @@
     "docker": "docker-compose build --no-cache && docker-compose run node"
```
Comment (bahaagalal) [inline package.json]: remove `--no-cache` and put `docker` script first.

File: package.json [hunk 11]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (bahaagalal) [inline package.json]: update `.eslintrc.json` to use 2 spaces indentation.

**Pull request merged**